### PR TITLE
Merge projects to single repository

### LINKED ISSUE #76: Single repository?
This ticket represents an idea I had, that I'd like to open up to the Gotham community to consider pros and cons.

I was reviewing https://github.com/gotham-rs and it strikes me that even at this early stage we have a number of repositories with a lot of duplication on things like licenses, contribution guidelines, issue triage etc.

There may be some way we can reasonably combine most, if not all of these into a single repository (this one) which may (or may not!) make for easier long term management of the project.

The structure for a single repository is currently unclear to me but it would need to making working with the following accessible for both people and tools:

* The `gotham` and `gotham_derive` crates, via `cargo`;
* Additional crates, such as [middlewares](https://github.com/gotham-rs/middleware-diesel) that are being maintained by the Gotham core team, again via `cargo`;
* Documentation, especially that which is published as [the Gotham book](https://github.com/gotham-rs/book);
* Community policies; and
* The [Gotham website](https://github.com/gotham-rs/gotham.rs), which is hosted by GitHub.

All comments welcome.

### PR DESCRIPTION
**Pull request by bradleybeddoes**
There ended up being three phases with this PR

1. Repository merges
Combines the following repositories into the core Gotham repository (these will subsequently be removed once 0.2 ships and marked archived once this PR lands):

* https://github.com/gotham-rs/borrow-bag
* https://github.com/gotham-rs/middleware-diesel
* https://github.com/gotham-rs/middleware-template
* https://github.com/gotham-rs/hello-world
* https://github.com/gotham-rs/policies

Deprecates the following repositories (these will subsequently be removed once 0.2 ships and marked archived once this PR lands):

* https://github.com/gotham-rs/example-app

2. Deprecating kitchen-sink 
The directory `examples/kitchen-sink` has been removed. Subsequent PR (so each example can be correctly scrutinised) will add back in a number of example crates that will allow developers building Gotham backed apps to succulently explore all the various components that Gotham offers without a singular, over whelming chunk of code

3. Cleaning up README/policies
The intent of these has been kept but there was a lot of extraneous wording that we just didn't need. In the case of the main Gotham README I have also added some best practice sub headings including listing alternate projects.

Closes #76.

### CODE REVIEW

**Comment by ChristophWurst** [comment]
> Disable travis cargo cache

Why is that necessary? I assume that will slow down builds a lot.

**Comment by bradleybeddoes** [comment]
@ChristophWurst errors like this started occurring https://travis-ci.org/gotham-rs/gotham/jobs/326651995#L784

A brief perusal doesn't show times being impacted much, if at all, based on the last few successful builds prior to that change.

**Comment by ChristophWurst** [comment]
I wonder if that's just a temporary issue and clearing the cache once would have helped.

File: README.md [hunk 1]
```diff
@@ -1,44 +1,62 @@
...
-The Gotham core team loves many of the elegant concepts that are found in dynamically typed web application frameworks, such as Rails, Phoenix and Django and aspire to achieve them with the type and memory safety guarantees provided by Rust.
-
-Gotham is stability focused. All releases of Gotham are compatible with stable Rust and that contract will never change. To ensure we remain compatible, we also regularly build against beta and nightly Rust.
-
-Gotham leverages async extensively thanks to the [Tokio project](https://tokio.rs) and is further enhanced by being built directly on top of async [Hyper](https://hyper.rs). Simple Gotham request handlers are able to complete in **µs** with very low memory requirements.
+1.  Stability focused. All releases target **stable** Rust. This will never change.
+    To ensure future compatibility, we also run automated builds against Rust beta and
+    nightly releases.
+1.  Statically typed. Unlike other web frameworks Gotham is statically typed ensuring your
+    entire application is **correctly expressed** at compile time.
```
Comment (smangelsdorf) [inline README.md]: > Unlike other web frameworks

Might be unnecessary

File: Cargo.toml [hunk 2]
```diff
@@ -1,43 +1,12 @@
...
-keywords = ["http", "async", "web", "framework", "gotham"]
+[workspace]
+members = [
+    "gotham",
+    "gotham_derive",
 
-[features]
-default = []
-ci = []
+		"middleware/template",
```
Comment (smangelsdorf) [inline Cargo.toml]: Nit: Is this a tab?

File: examples/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Gotham Examples
+
+Standalone crates that act as examples for building web applications with Gotham.
+
+All include test cases that prove correctness and serve as an example of how to test your own applications.
```
Comment (smangelsdorf) [inline examples/README.md]: > prove correctness

"demonstrate correct behaviour" perhaps? 😛

File: examples/hello_world/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    );
+
+    (state, res)
+}
+
+/// Start a server and call the `Handler` we've defined above for each `Request` we receive.
+pub fn main() {
+    let addr = "127.0.0.1:7878";
+    println!("Listening for requests at http://{}", addr);
+    start(addr, || Ok(say_hello))
```
Comment (smangelsdorf) [inline examples/hello_world/src/main.rs]: Not a fan of keeping this explicitly as `gotham::start` in code? I think importing the bare function makes it a bit harder to see what's going on.

File: Cargo.toml (hunk 2 above)
Comment (bradleybeddoes) [reply Cargo.toml]: Fixed.

File: README.md (hunk 1 above)
Comment (bradleybeddoes) [reply README.md]: Copied from website but agree, fixed.

File: examples/README.md (hunk 3 above)
Comment (bradleybeddoes) [reply examples/README.md]: fixed

File: examples/hello_world/src/main.rs (hunk 4 above)
Comment (bradleybeddoes) [reply examples/hello_world/src/main.rs]: Agree, `gotham::start` more useful, especially for newcomers.

Comment (bradleybeddoes) [reply examples/hello_world/src/main.rs]: fixed

**Comment by smangelsdorf** [comment]
Something weird going on with Coveralls, so I've temporarily removed the requirement for Coveralls to report everything is ok.

**Pull request merged**